### PR TITLE
Enable UT after thread qt fix part2

### PR DESCRIPTION
**Pull request by OHerasym**
### PR Changes:
- Enable policy tests after thread qt fix
- Enable protocol_handler tests after thread qt fix
- Enable&fix rpc_tests after thread qt fix
- Enable security_manager tests after thread qt fix
- Enable transport_manager tests after thread qt fix

> Tests builds on LINUX & WINDOWS & QT platform

Related [ticket](https://adc.luxoft.com/jira/browse/APPLINK-23755)

> Disabled tests after enabling failed qt tests
#### Disabled policy tests(25 tests):
- usage_statistics_test.cc:136:    DISABLED_AppStopwatchStartMethod_CallONCE_StatisticsManagerAddMethodCalledONCE) {
- usage_statistics_test.cc:150:     DISABLED_AppStopwatchStartMethod_Call_StatisticsManagerAddMethodCALLED) {
- usage_statistics_test.cc:166:     DISABLED_AppStopwatchSwitchMethod_Call_StatisticsManagerAddMethodCALLED) {
- usage_statistics_test.cc:182:    DISABLED_AppStopwatchStartMethod_CallAnd1SecSleepAfter_StatisticsManagerAddMethodCalledWith1SecTimespan) {
- usage_statistics_test.cc:197:    DISABLED_AppStopwatchSwitchMethod_CallAnd1SecSleepAfter_StatisticsManagerAddMethodCalledWith1SecTimespan) {
- shared_library_test.cc:51:    DISABLED_FullTest_OpenLibrarySetSymbolCloseLibrary_ExpectActsWithoutErrors) {
- policy_manager_impl_test.cc:430:TEST_F(PolicyManagerImplTest2, DISABLED_GetNotificationsNumberAfterPTUpdate) {
- policy_manager_impl_test.cc:468:       DISABLED_IsAppRevoked_SetRevokedAppID_ExpectAppRevoked) {
- policy_manager_impl_test.cc:581:       DISABLED_KmsChanged_SetExceededKms_ExpectCorrectSchedule) {
- policy_manager_impl_test.cc:597:    DISABLED_AddApplication_AddNewApplicationFromDeviceWithoutConsent_ExpectUpdateRequired) {
- policy_manager_impl_test.cc:607:    DISABLED_AddApplication_AddExistingApplicationFromDeviceWithoutConsent_ExpectNoUpdateRequired) {
- policy_manager_impl_test.cc:621:       DISABLED_PTUpdatedAt_DaysNotExceedLimit_ExpectNoUpdateRequired) {
- policy_manager_impl_test.cc:642:TEST_F(PolicyManagerImplTest2, DISABLED_ForcePTExchange_ExpectUpdateNeeded) {
- policy_manager_impl_test.cc:652:TEST_F(PolicyManagerImplTest2, DISABLED_OnSystemReady) {
- policy_manager_impl_test.cc:660:TEST_F(PolicyManagerImplTest2, DISABLED_ResetRetrySequence) {
- policy_manager_impl_test.cc:669:TEST_F(PolicyManagerImplTest2, DISABLED_NextRetryTimeout_ExpectTimeoutsFromPT) {
- policy_manager_impl_test.cc:694:TEST_F(PolicyManagerImplTest2, DISABLED_TimeOutExchange) {
- policy_manager_impl_test.cc:701:TEST_F(PolicyManagerImplTest2, DISABLED_GetPolicyTableStatus_ExpectUpToDate) {
- policy_manager_impl_test.cc:709:       DISABLED_RetrySequenceDelaysSeconds_Expect_CorrectValues) {
- policy_manager_impl_test.cc:730:       DISABLED_OnExceededTimeout_GetPolicyTableStatus_ExpectUpdateNeeded) {
- policy_manager_impl_test.cc:740:       DISABLED_GetInitialAppData_ExpectReceivedConsentCorrect) {
- policy_manager_impl_test.cc:783:    DISABLED_CanAppKeepContext_SetPoliciesForAppUpdated_ExpectAppCanKeepContext) {
- policy_manager_impl_test.cc:795:    DISABLED_CanAppStealFocus_SetPoliciesForAppUpdated_ExpectAppCanStealFocus) {
- policy_manager_impl_test.cc:804:TEST_F(PolicyManagerImplTest2, DISABLED_GetCurrentDeviceId) {
- policy_manager_impl_test.cc:812:       DISABLED_GetVehicleInfo_SetVehicleInfo_ExpectReceivedInfoCorrect) {
- sql_pt_representation_test.cc:404:       DISABLED_OpenAttemptTimeOut_ExpectCorrectNumber) {
- sql_pt_representation_test.cc:1029:     DISABLED_Init_TryInitNotExistingDataBase_ExpectResultFail) {
- sql_pt_representation_test.cc:1042:     DISABLED_Close_InitNewDataBaseThenClose_ExpectResultSuccess) {
- sql_pt_representation_test.cc:1464:TEST(SQLPTRepresentationTest3, DISABLED_RemoveDB_RemoveDB_ExpectFileDeleted) {
- sql_pt_representation_test.cc:1483:       DISABLED_GenerateSnapshot_SetPolicyTable_SnapshotIsPresent) {
- sql_pt_representation_test.cc:1524:       DISABLED_Save_SetPolicyTableThenSave_ExpectSavedToPT) {
#### Disabled protocol_handler tests(7 tests):
- protocol_packet_test.cc:213:TEST_F(ProtocolPacketTest, DISABLED_DeserializeZeroPacket) {
- protocol_handler_tm_test.cc:1072:TEST_F(ProtocolHandlerImplTest, DISABLED_MalformedLimitVerification) {
- protocol_handler_tm_test.cc:1113:       DISABLED_MalformedLimitVerification_MalformedStock) {
- protocol_handler_tm_test.cc:1234:       DISABLED_MalformedLimitVerification_NullTimePeriod) {
- protocol_handler_tm_test.cc:1262:TEST_F(ProtocolHandlerImplTest, DISABLED_MalformedLimitVerification_NullCount) {
- protocol_payload_test.cc:234:     DISABLED_ExtractProtocolWithJSONWithDataWithWrongPayloadSize) {
- multiframe_builder_test.cc:421:TEST_F(MultiFrameBuilderTest, DISABLED_Add_ConsecutiveFrames_per1) {
#### Disabled rpc_base tests (6 tests):
- rpc_base_json_test.cc:91:TEST(ValidatedTypesJson, DISABLED_BooleanFromInvalidJsonTest) {
- rpc_base_json_test.cc:98:TEST(ValidatedTypesJson, DISABLED_IntegerFromJsonTest) {
- rpc_base_json_test.cc:138:TEST(ValidatedTypesJson, DISABLED_IntegerFromOutOfRangeValueTest) {
- rpc_base_json_test.cc:185:TEST(ValidatedTypesJson, DISABLED_StringFromInvalidJsonTest) {
- rpc_base_json_test.cc:349:TEST(ValidatedTypesJson, DISABLED_NullableIntFromNonNullValueTest) {
- rpc_base_json_test.cc:367:TEST(ValidatedTypesJson, DISABLED_OptionalIntFromJsonTest) {
#### Disabled security tests(23 tests):
- ssl_certificate_handshake_test.cc:355:TEST_F(SSLHandshakeTest, DISABLED_CAVerification_ServerSide) {
- ssl_certificate_handshake_test.cc:372:TEST_F(SSLHandshakeTest, DISABLED_CAVerification_ServerSide_NoCACertificate) {
- ssl_certificate_handshake_test.cc:399:TEST_F(SSLHandshakeTest, DISABLED_CAVerification_ClientSide) {
- ssl_certificate_handshake_test.cc:416:TEST_F(SSLHandshakeTest, DISABLED_CAVerification_ClientSide_NoCACertificate) {
- ssl_certificate_handshake_test.cc:445:TEST_F(SSLHandshakeTest, DISABLED_CAVerification_BothSides) {
- ssl_certificate_handshake_test.cc:479:TEST_F(SSLHandshakeTest, DISABLED_ExpiredCert) {
- ssl_certificate_handshake_test.cc:497:TEST_F(SSLHandshakeTest, DISABLED_AppNameAndAppIDInvalid) {
- ssl_certificate_handshake_test.cc:550:TEST_F(SSLHandshakeTest, DISABLED_CAVerification_BothSides_ResetConnection) {
- security_query_test.cc:138:TEST_F(SecurityQueryTest, DISABLED_QueryConstructor) {
- security_query_test.cc:156:TEST_F(SecurityQueryTest, DISABLED_QueryConstructor2) {
- security_query_test.cc:223:TEST_F(SecurityQueryTest, DISABLED_Parse_NullData) {
- security_query_test.cc:238:TEST_F(SecurityQueryTest, DISABLED_Parse_LessHeaderData) {
- security_query_test.cc:255:TEST_F(SecurityQueryTest, DISABLED_Parse_HeaderData) {
- security_query_test.cc:276:TEST_F(SecurityQueryTest, DISABLED_Parse_HeaderDataWrong) {
- security_query_test.cc:300:TEST_F(SecurityQueryTest, DISABLED_Parse_InvalidQuery) {
- security_query_test.cc:336:TEST_F(SecurityQueryTest, DISABLED_Parse_InvalidQuery_UnknownTypeId) {
- security_query_test.cc:364:TEST_F(SecurityQueryTest, DISABLED_Parse_InvalidQuery_UnknownId_Response) {
- security_query_test.cc:389:TEST_F(SecurityQueryTest, DISABLED_Parse_Handshake) {
- security_query_test.cc:423:TEST_F(SecurityQueryTest, DISABLED_Parse_InternalError) {
- ssl_context_test.cc:365:TEST_F(SSLTest, DISABLED_OnTSL2Protocol_BrokenHandshake) {
- ssl_context_test.cc:384:TEST_F(SSLTest, DISABLED_OnTSL2Protocol_Positive) {
- ssl_context_test.cc:439:TEST_F(SSLTest, DISABLED_OnTSL2Protocol_EcncryptionFail) {
- crypto_manager_impl_test.cc:137:TEST_F(CryptoManagerTest, DISABLED_WrongInit) {
#### Disabled transport_manager tests(20 tests):
- transport_manager_default_test.cc:45:TEST(TestTransportManagerDefault, DISABLED_Init_LastStateNotUsed) {
- transport_manager_default_test.cc:59:TEST(TestTransportManagerDefault, DISABLED_Init_LastStateUsed) {
- transport_manager_impl_test.cc:336:TEST_F(TransportManagerImplTest, DISABLED_SearchDevices_AdaptersNotAdded) {
- transport_manager_impl_test.cc:344:TEST_F(TransportManagerImplTest, DISABLED_AddTransportAdapterSecondTime) {
- transport_manager_impl_test.cc:349:TEST_F(TransportManagerImplTest, DISABLED_ConnectDevice) {
- transport_manager_impl_test.cc:471:TEST_F(TransportManagerImplTest, DISABLED_SearchDevices_DeviceNotFound) {
- transport_manager_impl_test.cc:480:TEST_F(TransportManagerImplTest, DISABLED_SearchDevices_AdapterNotSupported) {
- transport_manager_impl_test.cc:489:TEST_F(TransportManagerImplTest, DISABLED_SearchDevices_AdapterWithBadState) {
- transport_manager_impl_test.cc:498:TEST_F(TransportManagerImplTest, DISABLED_SendMessageToDevice) {
- transport_manager_impl_test.cc:664:       DISABLED_ReceiveEventFromDevice_OnSearchDeviceDone) {
- transport_manager_impl_test.cc:683:       DISABLED_ReceiveEventFromDevice_OnSearchDeviceFail) {
- transport_manager_impl_test.cc:702:       DISABLED_ReceiveEventFromDevice_DeviceListUpdated) {
- transport_manager_impl_test.cc:888:TEST_F(TransportManagerImplTest, DISABLED_HandleMessage_ConnectionNotExist) {
- transport_manager_impl_test.cc:906:       DISABLED_SetVisibilityOn_TransportAdapterNotSupported) {
- transport_manager_impl_test.cc:913:       DISABLED_SetVisibilityOff_TransportAdapterNotSupported) {
- tcp_transport_adapter_test.cc:74:TEST_F(TcpAdapterTest, DISABLED_StoreDataWithOneDeviceAndOneApplication) {
- tcp_transport_adapter_test.cc:116:TEST_F(TcpAdapterTest, DISABLED_StoreDataWithSeveralDevicesAndOneApplication) {
- tcp_transport_adapter_test.cc:171:       DISABLED_StoreDataWithSeveralDevicesAndSeveralApplications) {
- tcp_transport_adapter_test.cc:272:TEST_F(TcpAdapterTest, DISABLED_StoreDataWithOneDevice_RestoreData) {
- tcp_transport_adapter_test.cc:307:TEST_F(TcpAdapterTest, DISABLED_StoreDataWithSeveralDevices_RestoreData) {

### CODE REVIEW

**Comment by OHerasym** [comment]
Please review: @anosach-luxoft , @dev-gh

File: src/components/transport_manager/test/CMakeLists.txt [hunk 1]
```diff
@@ -69,8 +69,8 @@ endif()
 set(SOURCES
   ${TM_TEST_DIR}/transport_manager_default_test.cc
   #TODO(OHerasym) : last_state assert fails
```
Comment (anosach-luxoft) [inline src/components/transport_manager/test/CMakeLists.txt]: @OHerasym please remove

File: src/components/protocol_handler/test/CMakeLists.txt [hunk 2]
```diff
@@ -51,7 +51,7 @@ set (SOURCES
   incoming_data_handler_test.cc
   protocol_header_validator_test.cc
   #TODO(OHerasym) : test don't finish executing
```
Comment (anosach-luxoft) [inline src/components/protocol_handler/test/CMakeLists.txt]: @OHerasym please remove

File: src/components/rpc_base/test/rpc_base_test.cc [hunk 3]
```diff
@@ -51,6 +53,17 @@ bool IsValidEnum(TestEnum val) {
   return val == kValue0 || val == kValue1;
 }
 
+void CompactJson(std::string& str) {
```
Comment (anosach-luxoft) [inline src/components/rpc_base/test/rpc_base_test.cc]: @OHerasym please use `CompactJson` from `formatters`

File: src/components/transport_manager/test/CMakeLists.txt (hunk 1 above)
Comment (OHerasym) [inline src/components/transport_manager/test/CMakeLists.txt]: @anosach-luxoft , Fixed

File: src/components/protocol_handler/test/CMakeLists.txt (hunk 2 above)
Comment (OHerasym) [inline src/components/protocol_handler/test/CMakeLists.txt]: @anosach-luxoft , Fixed

File: src/components/rpc_base/test/rpc_base_test.cc (hunk 3 above)
Comment (OHerasym) [inline src/components/rpc_base/test/rpc_base_test.cc]: @anosach-luxoft , Fixed

**Pull request merged**